WordPress/gutenberg
# Editor: Factor initial edits to own state

Extracted from / complements: #10844
Partially addresses: #7409

This pull request seeks to refactor the treatment of "initial edits" out of the `state.editor.present.edits` to allow for the existing `edits` state to represent only user-initiated edits to be considered for change detection. This will enable some simplification of change detection reset conditions, and works toward #10844's goal of considering "dirtiness" as a condition of a non-empty `edits` state.

**Implementation notes:**

While I think this partly moves away from a goal of eliminating the editor setup actions, it concedes in acknowledging that there should be at most a single parse which occurs at the start of an editing session, with consideration of both edits and content. Thus, it is implemented here as part of the effect handler for `SETUP_EDITOR`. I still have some faith this could be refactored in the future, but have chosen the most direct option to supporting #10844 here.

**Testing instructions:**

Verify there are no regressions in the behavior of "initial edits", notably:

- Saving a new draft with an empty title does not show "Auto Draft" upon reload
- The demo content works as expected

Ensure tests pass:

```
npm test
```

**Follow-up tasks:**

- [ ] With improvements here to initial edits, [assignment of draft status during first auto-draft save](https://github.com/WordPress/gutenberg/blob/3600c88fda01b62463769b5e8c3e795b064e12c1/packages/editor/src/store/effects/posts.js#L69-L82) could be easily moved to the server-side setup of a new draft.
- [ ] `SETUP_EDITOR` handling of blocks should be eliminated in favor of a consistent, single `RESET_BLOCKS` action.
